pallets/attestation/src/lib.rs

```diff
@@ -465,33 +479,41 @@ pub mod pallet {
 	}
 
 	impl<T: Config> Pallet<T> {
-		fn remove_attestation(attestation: AttestationDetails<T>, claim_hash: ClaimHashOf<T>) {
-			kilt_support::free_deposit::<AccountIdOf<T>, CurrencyOf<T>>(&attestation.deposit);
+		fn remove_attestation(attestation: AttestationDetails<T>, claim_hash: ClaimHashOf<T>) -> DispatchResult {
 			Attestations::<T>::remove(claim_hash);
+			AttestationStorageDepositCollector::<T>::free_deposit(attestation.deposit)?;
```

Comment:
Any reason why we don't first try to free the deposit, and then removing the attestation?

Reply:
Changed order: [38982fa](https://github.com/KILTprotocol/kilt-node/pull/527/commits/38982fafb3b85d7f238e93028c70bc1ad5284b75)